File: AlvaroToledo_GenerarTiempos.py
import matplotlib.pyplot as plt
import numpy as np
import time

#Variables
N=35

#Funcion de Fibonacci
def fibonacci(N):
	if N==0:
		return 0	
	if N==1:
		return 1
	else: 
		return fibonacci(N-1)+fibonacci(N-2)

#Funcion que mide el tiempo de la realizacion de Fibonacci
def get_time (N):
	t0 = time.time()
	#Corremos fibonacci(N)
	fibonacci(N)
	t1 = time.time()-t0
	return t1

#Imprimipos el timpo para valores de N
for i in range(N+1):
	print i, get_time(i)

File: AlvaroToledo_GenerarTiempos.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
using namespace std;
// Declaramos las funciones
int fibonacci (int N);
double get_time(int N);

//Main del codigo
int main(){
	int N = 35;
	for (int i=0; i<=N; i++){
		cout << i << " "<< get_time(i) << endl;
	}
	return 0;
}


//Sucesion de Fibonacci
int fibonacci (int N) {
	if (N==0){
		return 0;	
	}
	if (N==1){
		return 1;
	}
	else {
		return fibonacci(N-1)+fibonacci(N-2);	
	}
}

//Tiempo que se tarda en hacer la sucesion de Fibonacci

double get_time(int N){
	clock_t t;
	t = clock();
	// Corremos fibonacci(N)
	fibonacci(N); 
	t = clock() - t;
	float time = ((float)t)/CLOCKS_PER_SEC;
	return time;
}


